### PR TITLE
Clean up examples `package.json`

### PR DESCRIPTION
**Pull request by lucleray**
Clean up package.json files in the `examples` directory:
- Add `private: true`
- Remove `version` (because they are irrelevant for packages that are not meant to be published)
- Remove `name` (because they are optional for packages that are not meant to be published, and when someone clones an example, they often rename it and the property becomes stale)
- Remove `author`
- Remove `description`
- Remove `license`

Also remove `with-dynamic-app-layout` example completely, since it does the same as `layout-component` (https://github.com/vercel/next.js/pull/27121#discussion_r668178408).

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

**Review by styfle — DISMISSED** [review]
Looks good, thanks! 🎉 

It would be nice to add a lint step to make sure the package.json follows this pattern for new examples in the future 👍

**Review by ijjk — DISMISSED** [review]

File: examples/with-react-bootstrap/package.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
-  "name": "with-react-bootstrap",
-  "version": "0.1.0",
+  "private": true,
   "author": "Wenzhi Li",
```
Comment (leerob) [inline examples/with-react-bootstrap/package.json:3]:
```suggestion
```

File: examples/with-stencil/packages/test-component/package.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
-  "name": "test-component",
-  "version": "0.0.1",
+  "private": true,
   "description": "Stencil Component Starter",
```
Comment (leerob) [inline examples/with-stencil/packages/test-component/package.json:3]:
```suggestion
```

File: examples/with-patternfly/package.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
-  "name": "with-patternfly",
+  "private": true,
   "author": "Daniel Reinoso <danielreinoso1807@gmail.com>",
```
[the remark this answers is not in the document]
Comment (lucleray) [reply examples/with-patternfly/package.json:3]: Do we want to remove all authors?

[the remark this answers is not in the document]
Comment (leerob) [reply examples/with-patternfly/package.json:3]: I think so. It does seem a bit strange if you `npx create-next-app` to have their name in your repo. Their contribution is still preserved through git history. Thoughts?

[the remark this answers is not in the document]
Comment (leerob) [reply examples/with-patternfly/package.json:3]: Also, often multiple people are updating / contributing to these - not one single person.

[the remark this answers is not in the document]
Comment (lucleray) [reply examples/with-patternfly/package.json:3]: I agree 👍

File: examples/with-dynamic-app-layout/package.json [hunk 4]
```diff
@@ -1,8 +1,5 @@
 {
-  "name": "with-dynamic-app-layout",
```
[the remark this answers is not in the document]
Comment (lucleray) [reply examples/with-dynamic-app-layout/package.json:2]: Done in https://github.com/vercel/next.js/pull/27121/commits/e3a149b29163b78668e49ab474ea3eb22845c3d2.

File: examples/blog/package.json [hunk 5]
```diff
@@ -1,7 +1,5 @@
 {
-  "name": "portfolio",
-  "version": "1.0.0",
-  "main": "index.js",
+  "private": true,
   "license": "MIT",
```
[the remark this answers is not in the document]
Comment (lucleray) [reply examples/blog/package.json]: Removed all `license` in https://github.com/vercel/next.js/pull/27121/commits/3dcb42878b6cf4d5d9a55d3a0936f34d498e4365.

File: examples/with-patternfly/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (lucleray) [reply examples/with-patternfly/package.json]: Removed authors in https://github.com/vercel/next.js/pull/27121/commits/fee1906dd7f08429da40c98a6c6ec4a0cd0406dd.

**Review by leerob — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Pull request merged**